gadget/mkfs_test.go

```diff
@@ -0,0 +1,194 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package gadget_test
+
+import (
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/gadget"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type mkfsSuite struct {
+	cleanup []func()
+}
+
+var _ = Suite(&mkfsSuite{})
+
+func (m *mkfsSuite) SetUpTest(c *C) {
+	// fakeroot, mkfs.ext4, mkfs.vfat and mcopy are commonly installed in
+	// the host system, set up some overrides so that we avoid calling the
+	// host tools
+	cmdFakeroot := testutil.MockCommand(c, "fakeroot", "echo 'override in test' ; exit 1")
+	m.cleanup = append(m.cleanup, func() { cmdFakeroot.Restore() })
+
+	cmdMkfsExt4 := testutil.MockCommand(c, "mkfs.ext4", "echo 'override in test' ; exit 1")
+	m.cleanup = append(m.cleanup, func() { cmdMkfsExt4.Restore() })
+
+	cmdMkfsVfat := testutil.MockCommand(c, "mkfs.vfat", "echo 'override in test'; exit 1")
+	m.cleanup = append(m.cleanup, func() { cmdMkfsVfat.Restore() })
+
+	cmdMcopy := testutil.MockCommand(c, "mcopy", "echo 'override in test'; exit 1")
+	m.cleanup = append(m.cleanup, func() { cmdMcopy.Restore() })
+}
+
+func (m *mkfsSuite) TearDownTest(c *C) {
+	for _, r := range m.cleanup {
```

Comment:
(nitpick) might be easier to just use testutil.BaseTest and m.AddClean(cmdFakeroot.Restore) there